sdk/python/feast/registry.py

```diff
@@ -104,20 +114,60 @@ def __init__(self, registry_config: RegistryConfig, repo_path: Path):
             repo_path: Path to the base of the Feast repository
             or where it will be created if it does not exist yet.
         """
-        registry_store_type = registry_config.registry_store_type
-        registry_path = registry_config.path
-        if registry_store_type is None:
-            cls = get_registry_store_class_from_scheme(registry_path)
-        else:
-            cls = get_registry_store_class_from_type(str(registry_store_type))
 
-        self._registry_store = cls(registry_config, repo_path)
-        self.cached_registry_proto_ttl = timedelta(
-            seconds=registry_config.cache_ttl_seconds
-            if registry_config.cache_ttl_seconds is not None
-            else 0
+        if registry_config:
+            registry_store_type = registry_config.registry_store_type
+            registry_path = registry_config.path
+            if registry_store_type is None:
+                cls = get_registry_store_class_from_scheme(registry_path)
+            else:
+                cls = get_registry_store_class_from_type(str(registry_store_type))
+
+            self._registry_store = cls(registry_config, repo_path)
+            self.cached_registry_proto_ttl = timedelta(
+                seconds=registry_config.cache_ttl_seconds
+                if registry_config.cache_ttl_seconds is not None
+                else 0
+            )
+
+    @classmethod
+    def from_proto(cls, regsitry_proto: RegistryProto):
```

Comment:
nit: `regsitry` is mispelled

Reply:
q: why do we need a `from_proto` for the registry? I thought the registry contents would always be persisted by the registry store

to clarify, even when we want to compare two registries, we will be comparing protos; I don't think we'll need to reconstruct a registry from a proto?

Reply:
yeah fair, I can change this to operate on the protos themselves. I think I misunderstood earlier